src/magic_folder/downloader.py

```diff
@@ -304,6 +304,12 @@ def _process(self, snapshot):
                           capability=snapshot.capability,
                           ) as action:
             relpath = magic2path(snapshot.name)
+
+            # if we're already conflicted, no further processing
+            if self._config.list_conflicts_for(snapshot.name):
+                action.add_success_fields(is_conflict=True)
+                return
```

Comment:
I don't think this is correct.

There are at least three cases:
- we are in conflict with `device-a` and `device-b`
- we are in conflict with `device-a` but `device-b` has a descendant of our version of the file.
- we are in conflict with `device-a` but have already written a conflict file, and `device-a` now has a newer version

I think in the first two cases we still want to proceed with the logic of this function. In the third, we probably want to update the conflict file *only* if the file matches the file we wrote (which would require us to track the snapshot we wrote to the conflict file). 

We could also choose to *never* overwrite in the third case. But if we choose that option, we need to make sure that when resolving the conflict, we record as parent the version we wrote (so we'll need to write the snapshot cap to the db as well).

Reply:
We don't support multiple devices yet, so nothing to do here until then.
Thus, the only immediately interesting case is "in conflict with `device-a`" (which is equivalent to your first case above).

In this case we want do want to stop and do nothing else because we can't know what, if anything, the user has seen nor what they've done as far as resolving the conflict .. we only learn that when they use the HTTP API to say so or use the filesystem API to say so.

If we _update_ the conflict file after writing it the user may make a decision based on stale information (e.g. because they opened the conflict file when it first appeared). They still may take considerable time to decide what to do because this likely takes some out-of-band communication. This allows a huge window for them to "resolve" the conflict based on the stale information.

Reply:
That is to say: a) we don't support multiple users yet and; b) even if we did, we _don't_ want to ever "update" conflict-files that already got written.

If we did decide we want to give users a way to bump ahead our local notion of the conflict (i.e. update any existing conflict-files and possibly write new ones, if there are more) then it should be an explicit HTTP API along with corresponding CLI command(s).